### ISSUE TITLE
Release date not shown in nightly release

### ISSUE THREAD

**Issue by heidihoward** [tlaplus/tlaplus#771]
Would it be possible to get the release date in the first line of tlc output for the nighly release?

Currently, I see the following when I run tlc from the nighly:
```
TLC2 Version 2.18 of Day Month 20?? (rev: 27eace5)
```

**Comment by heidihoward** [comment]
That's handy and sufficient for my needs, thank you!

**Comment by ahelwer** [comment]
This is done by PR #832, but unfortunately I would recommend both that PR and this issue be closed because adding timestamps to builds is a classic way to stop builds from being reproducible. Not that we have reproducible builds now, but it's a good thing to keep an eye on.

**Comment by heidihoward** [comment]
Personally, I've vote for usability over reproducible builds in this particular setting but I don't feel strongly (git commit is sufficient for me). It might look better if `Day Month 20??` was removed entirely from the cli output for the nightly build. Feel free to close

**Comment by lemmy** [comment]
@ahelwer There is no consensus that we need/want reproducible builds.

**Comment by ahelwer** [comment]
I recall you were concerned about build reproducibility in relation to OS-varying line endings of JavaCC-generated code.

As an idea here, we could replace the `TLC2 Version 2.18 of Day Month 20?? (rev: 27eace5)` (where does 2.18 come from?) text with just:
```
The TLC finite model checker for TLA⁺ (rev: 27eace5)
https://github.com/tlaplus/tlaplus/commit/27eace5
```

**Comment by ahelwer** [comment]
Actually I thought of something even better - commits contain timestamps, so just use the timestamp from the commit instead of the timestamp from the system clock at build time! Probably more semantically useful than the build time even. As long as we normalize the commit timestamp to UTC the build should be reproducible.

**Comment by ahelwer** [comment]
So I've taken a liking to the idea of adding the commit time to the build & have been looking at this, and unfortunately [the git ant task](https://github.com/alx3apps/jgit-buildnumber) we use is unmaintained & hasn't seen an update in a decade, and also does not expose the git commit time. There are a few possible things I could do:

1. Fork the ant task we're using and modify it to emit the latest commit time, then update the `tlatools/org.lamport.tlatools/lib/jgit-buildnumber-ant-task-1.2.10.jar` file in our repo to use a jar built from my fork
2. Add a feature to the upstream [JGit](https://github.com/eclipse-jgit/jgit) library that Ant task library is wrapping; JGit added Ant tasks last year, but [only Init/Clone/Checkout/Add](https://github.com/eclipse-jgit/jgit/tree/master/org.eclipse.jgit.ant/src/org/eclipse/jgit/ant/tasks); I could add a GitCommitInfo task

The second one seems like a lot more work but it would be nice to take a dependency on an actual maintained & widely-used library instead of a one-off fork of a project that's been abandoned for a decade.

**Comment by lemmy** [comment]
> It might look better if `Day Month 20??` was removed entirely from the cli output for the nightly builds. Feel free to close

Agreed, it doesn't look great, but it does indicate that `tla2tools.jar` is a nightly build. The reason nothing better is displayed is that I don't want to maintain code to suppress or change this placeholder. Additionally, the placeholder can easily be found [everywhere](https://github.com/tlaplus/tlaplus/blob/dff77775a044a71ac6251b8a587ec7091c97acc8/toolbox/org.lamport.tla.toolbox.product.product/org.lamport.tla.toolbox.product.product.product#L11-L15) [it](https://github.com/tlaplus/tlaplus/blob/dff77775a044a71ac6251b8a587ec7091c97acc8/toolbox/org.lamport.tla.toolbox.product.standalone/plugin.xml#L33) [appears](https://github.com/tlaplus/tlaplus/blob/dff77775a044a71ac6251b8a587ec7091c97acc8/toolbox/org.lamport.tla.toolbox.product.standalone/src/org/lamport/tla/toolbox/ui/intro/ToolboxIntroPart.java#L434) when we prepare for an actual release.  By the way, a build identifier and timestamp are backed into tla2tools.jar's `META-INF/MANIFEST.MF` because there can be many builds of a single commit.

**Comment by ahelwer** [comment]
Markus would you prefer to have the git commit time or build time printed out? In my mind it makes the most sense to have the commit time, for the following reasons:

1. build reproducibility, which we may want in the future
2. for local builds, the actual time of jar creation is already captured in the file metadata
3. users probably want to know "how old is this version" not when it was built specifically, although for official releases this will be within a few minutes anyway

**Comment by ahelwer** [comment]
Per discussion in July community call, this feature is on hold until we have separated out the tools from the toolbox

**Comment by lemmy** [comment]
PR https://github.com/tlaplus/tlaplus/pull/1328 updates TLC’s startup banner for the rolling release to display a lexicographically sortable, human-readable timestamp (see below).

```shell
-> % java -jar dist/tla2tools.jar
TLC2 Version 2026.02.24.164202 (rev: feb0c43)
[...]
```

Please provide your comments (if any) directly in the PR.
